enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,435 @@
+
+The registry provisioner has logic that converts OLM’s registry+v1 bundle format into a plain bundle format. The OLM ClusterServiceVersion (CSV) resource can be decomposed into a set of core Kubernetes manifests (e.g. RBAC, Deployment, etc.). This conversion logic allows the registry provisioner to effectively act as a bridge between higher-level components (e.g. PO manager) that want to apply registry+v1 bundle contents, and still have the rukpak stack manage those contents.
+
+After the registry provisioner successfully unpacks the registry+v1 bundle, and converts that into a plain bundle format, the plain provisioner will attempt to apply those manifests. In the case that an individual manifest is invalid, and the Kubernetes API server rejects the creation of that manifest, then the plain provisioner will update the BundleDeployment resource to reflect that failed rollout.
+
+> Note: The plain provisioner has the current limitation around applying Bundle contents that will produce runtime issues.
+
+In the case that unpacked Bundle contents contain a Deployment manifest, and that manifest references an invalid container image, then the result would be an unavailable container process after being applied to the cluster. Despite the presence of runtime issues here, the BundleDeployment that manages those manifests would reflect a successful installation state. Addressing this limitation in the context of PO packages is discussed in more detail in subsequent sections.
+
+#### Proxying PO State
+
+```mermaid
+sequenceDiagram
+participant CVO
+participant ClusterOperator
+participant PO Manager
+CVO->>ClusterOperator: Watches
+PO Manager->>ClusterOperator: Manages
+PO Manager-->ClusterOperator: Aggregates PO Package State
+PO Manager->>PO Custom Resources: Watches
+PO Custom Resources->>PO Manager: Relays .Status
+BundleDeployments->>PO Custom Resources: Relays .Status
+```
+
+The PO manager will be responsible for acting as a proxy between the CVO, and the underlying PO packages during phase 0. This can be done by reconciling an "aggregate" ClusterOperator that reflects the overall state of installing PO packages. Future phases may require PO packages to reconcile OLM's OperatorCondition API, or the directly manage a ClusterOperator resource to express their health and overall state.
+
+In order to address the existing plain provisioner limitation, this proposal also recommends expanding the plain+v0 bundle format, and enabling users to define health checks through a dedicated metadata file. A valid metadata file may be named "bundle.yaml" so clients can know where to find this file.
```

Comment:
> In order to address the existing plain provisioner limitation

might clarify what limitation you mean here. I think you mean the limitation that the plain provisioner doesn't really know if the provision was successful or not?

Reply:
It looks like I've refactored the EP and changed some language around.

> Note: The plain provisioner has the current limitation around applying Bundle contents that will produce runtime issues. In the case that unpacked Bundle contents contain a Deployment manifest, and that manifest references an invalid container image, then the result would be an unavailable container process after being applied to the cluster. Despite the presence of runtime issues here, the BundleDeployment that manages those manifests would reflect a successful installation state. Addressing this limitation is out-of-scope for phase 0, and will be addressed in future phases.

That's now the current snippet that's outlined.